lib/devices/ios/xcode.js

```diff
@@ -29,18 +72,26 @@ xcode.getPath = function (cb) {
 
 xcode.getVersion = function (cb) {
   xcode.getPath(function (err, xcodePath) {
-    if (err  !== null) {
-      cb(err, null);
+    if (err === null && !_hasExpectedSubDir(xcodePath)) {
+      err = new Error("Expected a path ending with '" + XCODE_SUBDIR + "', but got " + xcodePath);
+    }
+    if (err !== null) {
+      return cb(err, null);
     } else {
-      var xcodebuildPath = path.resolve(xcodePath, "usr/bin/xcodebuild");
-      if (!fs.existsSync(xcodebuildPath)) {
-        cb(new Error("Could not get Xcode version: " + xcodebuildPath + " does not exist on disk."), null);
+      // we want to read the CFBundleShortVersionString from Xcode's plist.
+      // It should be in /[root]/XCode.app/Contents/
+      var plistPath = xcodePath.replace(XCODE_SUBDIR, "/Contents/Info.plist");
+      if (!fs.existsSync(plistPath)) {
+        return cb(new Error("Could not get Xcode version: " + plistPath + " does not exist on disk."), null);
       } else {
-        exec(JSON.stringify(xcodebuildPath) + ' -version', { maxBuffer: 524288, timeout: 3000 }, function (err, stdout) {
+        var cmd = "/usr/libexec/PlistBuddy -c 'Print CFBundleShortVersionString' " + plistPath;
+        exec(cmd, { maxBuffer: 524288, timeout: 3000 }, function (err, stdout) {
+          if (err !== null) return cb(err, null);
```

Comment:
our idiom is usually `if (err) return cb(err);`

1) in many cases `err` is undefined, not null, when it doesn't exist
2) if we're returning an error, we don't need to return a null value for the success object